src/Sylius/Component/Core/spec/Uploader/ImageUploaderSpec.php

```diff
@@ -46,8 +46,25 @@ function it_uploads_an_image($filesystem, $image)
         $image->hasFile()->willReturn(true);
         $image->getPath()->willReturn('foo.jpg');
 
-        $filesystem->delete(Argument::any())->shouldBeCalled();
-        $filesystem->write(Argument::any(), Argument::any())->shouldBeCalled();
+        $filesystem->has('foo.jpg')->willReturn(false);
+
+        $filesystem->delete(Argument::any())->shouldNotBeCalled();
+        $filesystem->write('foo.jpg', Argument::any())->shouldBeCalled();
+
+        $image->setPath(Argument::any())->shouldBeCalled();
+
+        $this->upload($image);
+    }
+
+    function it_replaces_an_image($filesystem, $image)
+    {
+        $image->hasFile()->willReturn(true);
+        $image->getPath()->willReturn('foo.jpg');
```

Comment:
Looking at the `ImageUploader` logic:

```php
if (!$image->hasFile()) {
    return;
}

if (null !== $image->getPath() && $this->has($image->getPath())) {
    $this->remove($image->getPath());
}

do {
    $hash = md5(uniqid(mt_rand(), true));
    $path = $this->expandPath($hash.'.'.$image->getFile()->guessExtension());
} while ($this->filesystem->has($path));

$image->setPath($path);

$this->filesystem->write(
    $image->getPath(),
    file_get_contents($image->getFile()->getPathname())
);
```

`Image::$file` contains a file being currently uplaoded, `Image::$path` contains a path to already existing file, so:

1. If there's no uploaded file, do nothing
2. *Remove currently uploaded file __if there is any__*
3. **Generate path for uploaded image**
4. Set this generated path on an image
5. Upload that image to the generated path

Basically, this spec is skipping that 3rd part and saves the file at the same path as the previous one, which is unlikely to happen.